probe/endpoint/resolver.go

```diff
@@ -0,0 +1,71 @@
+package endpoint
+
+import (
+	"log"
+	"net"
+	"time"
+
+	"github.com/bluele/gcache"
+)
+
+// Default cache length
+const RAddrCacheLen = 500
+
+const raddrBacklog = 1000
+const rAddrCacheExpiration = 30 * time.Minute
+
+// ReverseResolver is a caching, reverse resolver
+type ReverseResolver struct {
+	addresses chan string
+	cache     gcache.Cache
+}
+
+// NewReverseResolver starts a new reverse resolver that
+// performs reverse resolutions and caches the result.
+func NewReverseResolver(cacheLen int) *ReverseResolver {
+	r := ReverseResolver{
+		addresses: make(chan string, raddrBacklog),
+		cache:     gcache.New(cacheLen).LRU().Expiration(rAddrCacheExpiration).Build(),
+	}
+
+	go func() {
```

Comment:
Please pull this out into a separate function, like in https://github.com/weaveworks/scope/blob/master/probe/endpoint/conntrack.go#L78